x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/querydsl/query/RegexQuery.java

```diff
@@ -31,14 +37,18 @@ public String regex() {
         return regex;
     }
 
+    public Boolean caseInsensitive() {
+        return caseInsensitive;
+    }
+
     @Override
     public QueryBuilder asBuilder() {
-        return regexpQuery(field, regex);
+        return regexpQuery(field, regex).caseInsensitive(caseInsensitive);
     }
 
     @Override
     public int hashCode() {
-        return Objects.hash(field, regex);
+        return Objects.hash(caseInsensitive, field, regex);
```

Comment:
Really minor: I'd prefer to have the `caseInsensitive` last in order as defined in the class and the ctor.
(Same for equals)